api/services/uaaStrategy.js

```diff
@@ -33,4 +35,34 @@ function createUAAStrategy(options, verify) {
   return strategy;
 }
 
-module.exports = { createUAAStrategy };
+async function verifyUAAUser(accessToken, refreshToken, profile, uaaGroups) {
+  const { user_id: uaaId, email } = profile;
+
+  const client = new UAAClient(accessToken);
+  const isVerified = await client.verifyUserGroup(uaaId, uaaGroups);
+
+  if (!isVerified) {
+    return null;
+  }
+
+  const identity = await UAAIdentity.findOne({ where: { email } });
+
+  if (!identity) {
+    return null;
+  }
+
+  const user = await identity.getUser();
```

Comment:
maybe eager load the user when fetching the identity. Also, the user being null shouldn't be possible, we don't need to handle, just let it fail eventually since it would be a problem.

Reply:
Check one two.